spec/lib/miq_automation_engine/miq_ae_object_spec.rb

```diff
@@ -81,19 +81,64 @@ def value_match(value, xml_value)
     expect(result["vms"].length).to eq(2)
   end
 
-  it "#process_args_as_attributes with an array containing invalid entries" do
-    vm2 = FactoryGirl.create(:vm_vmware)
-    result = @miq_obj.process_args_as_attributes({"Array::vms" => "VmOrTemplate::#{@vm.id},fred::12,,VmOrTemplate::#{vm2.id}"})
-    expect(result["vms"]).to be_kind_of(Array)
-    expect(result["vms"].length).to eq(2)
-  end
+  describe "#process_args_as_attributes" do
+    let(:result) { @miq_obj.process_args_as_attributes("Array::my_values" => my_values) }
+
+    context "with an array containing invalid entries" do
+      let(:my_values) { "VmOrTemplate::#{@vm.id},fred::12,,VmOrTemplate::#{FactoryGirl.create(:vm_vmware).id}" }
+
+      it "stores the first value as a VM object" do
+        expect(result["my_values"].first).to be_kind_of(MiqAeMethodService::MiqAeServiceVm)
+      end
+
+      it "stores the second value as a Host object" do
+        expect(result["my_values"].second).to eq("fred::12")
```

Comment:
@mkanoor - @gmcculloug's implementation will allow for "fred::12" to get passed as datatype `fred` value `12` to `process_args_as_attributes`.

We had talked earlier of forcing `process_args_as_attributes` to raise an error if a datatype was invalid.

I prototyped some code to see how this would affect the the current specs. 

`CLASS_TYPES       = %w( boolean TrueClass FalseClass time Time symbol Symbol integer Fixnum float Float array String password )`

With the list of allowed datatypes as above

`raise MiqAeException::InvalidClass unless CLASS_TYPES.include?(datatype)`

This brought out other issues as there are sundry objects passed through `process_args_as_attributes` when the a new instance of the MiqAeEngine is instantiated.  

I'm wondering if @gmcculloug's implementation will allow for fewer legacy issues.